vendor/github.com/opencontainers/runc/libcontainer/cgroups/systemd/apply_systemd.go

```diff
@@ -296,12 +321,16 @@ func (m *Manager) Apply(pid int) error {
 	}
 
 	statusChan := make(chan string)
-	if _, err := theConn.StartTransientUnit(unitName, "replace", properties, statusChan); err != nil && !isUnitExists(err) {
+	if _, err := theConn.StartTransientUnit(unitName, "replace", properties, statusChan); err == nil {
+		select {
+		case <-statusChan:
+		case <-time.After(time.Second):
```

Comment:
@filbranden we hit an issue with this in production and @sjenning @vikaschoudhary16 need to push a revert.

"In https://github.com/opencontainers/runc/pull/1683 and follow on https://github.com/opencontainers/runc/pull/1754, we attempted to handle a race condition https://github.com/openshift/origin/issues/16246 by waiting for systemd to ack the creation of the transient unit created by StartTransientUnit().

This did solve the race but unfortunately started the code down a path in the systemd dbus library that I'm now convinced was never tested.

If you pass a channel to StartTransientUnit() it takes the jobListener lock in startJob() where it didn't before.  This causes the deadlock situation when even one systemd ack is missed (for yet unknown reasons)."

see: https://github.com/openshift/origin/pull/19043

Reply:
@filbranden yes, this has been a mess.  We ended up reverting both  opencontainers/runc#1683  and  opencontainers/runc#1754 in the vendored runc for openshift/origin v3.9 because of the deadlock we could encounter in the `coreos/go-systemd/dbus` package. See `startJob()` and `jobComplete()` w/ the `jobListener` lock.

https://bugzilla.redhat.com/show_bug.cgi?id=1558183#c5

Unfortunately, it looks like we dropped to ball getting that revert upstream.   The channel wait still exists in runc master, kube master and 1.10, and openshift/origin master as we just rebased on kube 1.10 and resynced our deps.

The whole "passing a channel to StartTransientUnit()" need to be rolled back everywhere.

Reply:
Ok sounds good, let's hold on this.

Before we revert everything... May I suggest we spend some time trying to understand whether there's a bug in `coreos/go-systemd/dbus` that we can fix instead?

I'll cc myself to the Bugzilla in RedHat.

Thanks!
Filipe

Reply:
This is fixed in opencontainers/runc#1781.